extension/chrome/settings/modules/add_key.ts

```diff
@@ -84,6 +84,8 @@ View.run(class AddKeyView extends View {
         return await Ui.modal.warning(e.message, Ui.testCompatibilityLink);
       } else if (e instanceof KeyCanBeFixed) {
         return await Ui.modal.error(`This type of key cannot be set as non-primary yet. Please write human@flowcrypt.com`, false, Ui.testCompatibilityLink);
+      } else if (e instanceof Error && e.message === 'Unsupported key type: unknown') {
+        return await Ui.modal.warning(`Pasted content does not appear to be a validly formatted OpenPGP key.\n\n(Unsupported key type: unknown)`);
```

Comment:
A question: should we rather introduce a new kind of Error that we could `instanceof` with?

> validly formatted OpenPGP key

No mention of S/MIME? Maybe `does not appear to be a validly formatted OpenPGP nor S/MIME key` ?

Reply:
> A question: should we rather introduce a new kind of Error that we could instanceof with?

sounds good, I'll do that

> No mention of S/MIME? Maybe does not appear to be a validly formatted OpenPGP nor S/MIME key ?

Here testing for private keys, that's why only OpenPGP.

However we have a customer very, very interested in importing their s/mime private keys too.

Reply:
But as far as parsing goes, you're right - it checks for pgp or s/mime. I thought it would be somewhat confusing if we said pgp or s/mime, and then if they proceeded to enter valid s/mime, we would error again because we don't actually support it.

But that should be super rare, and would actually be and interesting way to have users write us that they are interested in this, so I'll do that :)

Reply:
> However we have a customer very, very interested in importing their s/mime private keys too.
Good to hear! I imagine after adding private S/MIME keys this modal message could be easily overlooked later so why not fix it now :)
Thanks!